src/display/ui_state.rs

```diff
@@ -155,7 +163,7 @@ impl UIState {
         self.processes = processes;
         self.remote_addresses = remote_addresses;
         self.connections = connections;
-        self.total_bytes_downloaded = total_bytes_downloaded;
-        self.total_bytes_uploaded = total_bytes_uploaded;
+        self.total_bytes_downloaded = total_bytes_downloaded / self.utilization_data.len() as u128;
```

Comment:
I woke up this morning thinking if we should test for a division by zero here :)

Reply:
Nah we're iterating over it, looks good, sorry!

Reply:
Ah, right. This is one of those "can't happen" sort of things, but I definitely agree we should test for it anyway (code changes, etc.)
Thanks - I hope you didn't lose sleep over it ;P